api/go.mod

```diff
@@ -6,7 +6,7 @@ replace github.com/hashicorp/consul/sdk => ../sdk
 
 require (
 	github.com/google/go-cmp v0.5.9
-	github.com/hashicorp/consul/sdk v0.14.1
+	github.com/hashicorp/consul/sdk v0.4.1-0.20230821222840-992198e5f8c7
```

Comment:
Am I reading that version string correctly as `0.4.1`, down from `0.14.1`?

Maybe didn’t intend to push this?

Reply:
I think this is a result of using a commit sha when using `go get`, I assume Dan had to do this since `v0.15.0` isn't released yet. It does some funky things with the version numbers when you do that, but it's probably correct 🤔

Reply:
Makes sense, and that was my read too, but I was surprised by the prefix.

Looks like https://go.dev/ref/mod#pseudo-versions might explain it: I'm guessing we've had a bunch of "orphan" tag commits due to `replace` commenting in releases, and `0.4.1` is the most recent semver tag `go mod` could find that's a parent of this commit.

Regardless, I wasn't expecting the bump here just due to the change under `sdk/` - I was expecting a change there after cutting `v0.15.0` of `sdk`, so I think the second commit might have been for testing? (Or I'm confused!)

Reply:
Ah, this change was a desperate attempt at not needing to tag the change. I removed that commit now.